src/librustc/diagnostics.rs

```diff
@@ -168,6 +168,11 @@ match x {
 ```
 "##,
 
+E0010: r##"
+The value of static and const variables must be known at compile time, and they
+live for the entire lifetime of a program. They can not use allocation.
```

Comment:
that looks good to me, especially I think this error code is currently only emitted for uses of `box` (and not other methods of dynamic memory allocation, all of which would currently also be illegal to even attempt to encode).